src/app/features/tech-record/components/change-vehicle-type/change-vehicle-type.component.ts

```diff
@@ -0,0 +1,96 @@
+
+@Component({
+  selector: 'app-change-vehicle-type',
+  templateUrl: './change-vehicle-type.component.html',
+  styleUrls: ['./change-vehicle-type.component.scss']
+})
+export class ChangeVehicleTypeComponent implements OnInit, OnChanges {
+  constructor(
+    private technicalRecordService: TechnicalRecordService,
+    public globalErrorService: GlobalErrorService,
+    private store: Store<TechnicalRecordServiceState>,
+    private router: Router,
+    private route: ActivatedRoute,
+    public dfs: DynamicFormService,
+    private location: Location
+  ) {
+    this.technicalRecordService.selectedVehicleTechRecord$.pipe(take(1)).subscribe(data => (this.vehicleTechRecord = data));
+    this.technicalRecordService.editableTechRecord$.pipe(take(1)).subscribe(data => (this.currentTechRecord = data));
+  }
+
+  public currentTechRecord?: TechRecordModel;
+  public vehicleTechRecord?: VehicleTechRecordModel;
+  public form!: CustomFormGroup;
+
+  public template: FormNode = {
+    name: 'criteria',
+    type: FormNodeTypes.GROUP,
+    children: [
+      {
+        name: 'selectVehicleType',
+        label: 'Select a new vehicle type',
+        value: '',
+        type: FormNodeTypes.CONTROL
+      }
+    ]
+  };
+
+  ngOnChanges(changes: SimpleChanges): void {
+    this.globalErrorService.clearErrors();
+
+    const { vehicleTechRecord } = changes;
+
+    if (this.form && vehicleTechRecord?.currentValue && vehicleTechRecord.currentValue !== vehicleTechRecord.previousValue) {
+      this.form.patchValue(vehicleTechRecord.currentValue, { emitEvent: false });
+    }
+  }
+
+  get currentVrm(): string | undefined {
+    return this.vehicleTechRecord?.vrms.find(vrm => vrm.isPrimary === true)?.vrm;
+  }
+
+  get vehicleTypeOptions(): MultiOptions {
+    return getOptionsFromEnum(VehicleTypes).filter(type => type.value != this.currentTechRecord?.vehicleType);
+  }
+
+  ngOnInit(): void {
+    this.form = this.dfs.createForm(this.template) as CustomFormGroup;
+    if (this.vehicleTechRecord === undefined) {
+      this.navigateBack();
+    }
+  }
+
+  handleSubmitNewVehicleType(selectedVehicleType: VehicleTypes): void {
+    if (!selectedVehicleType || selectedVehicleType === this.vehicleTechRecord?.techRecord[0].vehicleType) {
+      this.globalErrorService.addError({ error: 'You must provide a new vehicle type', anchorLink: 'selectedVehicleType' });
+      return;
+    } else if (selectedVehicleType !== VehicleTypes.PSV) {
+      this.globalErrorService.addError({ error: 'That technical feature will be implemented soon', anchorLink: 'selectedVehicleType' });
+      return;
+    }
+    this.store.dispatch(changeVehicleType({ vehicleType: selectedVehicleType }));
+
+    this.currentTechRecord?.statusCode !== StatusCodes.PROVISIONAL
+      ? this.router.navigate(['../amend-reason'], { relativeTo: this.route })
+      : this.router.navigate(['../notifiable-alteration-needed'], { relativeTo: this.route });
```

Comment:
```suggestion
    const routeSuffix = this.currentTechRecord?.statusCode !== StatusCodes.PROVISIONAL ? 'amend-reason' : 'notifiable-alteration-needed';
    this.router.navigate([`../${routeSuffix}`], { relativeTo: this.route });
```

Reply:
Is this better? or just different?

Reply:
This portion of the code is duplicated on both lines
`this.router.navigate(['../* actual route */'], { relativeTo: this.route })`
and it does not need to be because the only thing that changes is the routeing suffix.

Reply:
I can address this later on, so I am resolving the comment.

Reply:
```
    this.currentTechRecord?.statusCode !== StatusCodes.PROVISIONAL
      ? this.navigateTo('../amend-reason')
      : this.navigateTo('../notifiable-alteration-needed');
```

I like this function